### PR TITLE
If HA, set the webhooks failure policy to 'Fail'

### LINKED ISSUE #2852: Support setting FailurePolicy of proxy injector webhook
## Feature Request
Support specifying the failure policy of the proxy injector webhook during `linkerd install` and `linkerd upgrade`.

### What problem are you trying to solve?
We'd like to prevent deployments of apps, especially when things go wrong (node on which the proxy injector webhook runs is partitioned from the rest of the cluster and a failing node's pods being rescheduled onto healthy nodes), to result in non-meshed applications.

### How should the problem be solved?
Optionally allow configuring the failure policy of the proxy injector webhook during the control plane installation/upgrade.

### Any alternatives you've considered?

We're using kustomize so we can just patch the webhook configuration manifest now that #2806 has landed but not everyone does so it'd be nice to be able to pass it.

### How would users interact with this feature?
```
linkerd install --proxy-injector-failure-policy=fail | kubectl apply -f -
```

### PR DESCRIPTION
**Pull request by alpeb**
I'm adding to the linkerd namespace a new label
`linkerd.io/is-control-plane: true` that is used in the webhook configs'
selector to skip the proxy injector for this namespace. This avoids
running into the timing issues described in #2852.

Fixes #2852

### CODE REVIEW

**Comment by ihcsim** [comment]
Any reasons why we won't want to make this our default? I think it makes sense for pods creation to fail, and not appear uninjected, if the proxy injector isn't reachable.

**Comment by grampelberg** [comment]
@ihcsim I was optimizing for the initial getting started experience. If that's still great with this turned on (like, it doesn't hose your entire cluster), then having a single default would be great.

**Comment by alpeb** [comment]
If a proxy injector pod dies, when k8s tries to create it again it will call the mutating webhook. If that was the only proxy injector pod (non-HA mode) then it'll fail and the policy might better be `Ignore` otherwise we would get stuck.

**Comment by ihcsim** [comment]
But we are already using the MWC namespace selector to ignore pod events in the `linkerd` namespace, right?

**Comment by alpeb** [comment]
> But we are already using the MWC namespace selector to ignore pod events in the `linkerd` namespace, right?

Oh right right, in that case it makes sense to leave `Fail` as the default always ;-)

File: cli/cmd/install.go [hunk 1]
```diff
@@ -511,6 +523,10 @@ func (options *installOptions) validate() error {
 		return errors.New("--proxy-log-level must not be empty")
 	}
 
+	return nil
+}
+
+func (options *installOptions) handleHA() {
```
[the remark this answers is not in the document]
Comment (alpeb) [reply cli/cmd/install.go]: I probably had a mishap when I rebased. Will rebase again.

File: cli/cmd/install.go [hunk 2]
```diff
@@ -379,11 +382,20 @@ func (options *installOptions) validateAndBuild(stage string, flags *pflag.FlagS
...
+
 	values.stage = stage
 
 	return values, configs, nil
 }
 
+func (options *installOptions) setWebhookFailurePolicy(values *installValues) {
+	values.WebhookFailurePolicy = "Ignore"
+	if options.highAvailability {
+		values.WebhookFailurePolicy = "Fail"
```
[the remark this answers is not in the document]
Comment (alpeb) [reply cli/cmd/install.go]: Given now we'll always use Fail, I've removed this function in my last push.

**Pull request merged**